src/EditorFeatures/Core/Extensibility/NavigationBar/WrappedNavigationBarItem.cs

```diff
@@ -33,18 +33,19 @@ internal WrappedNavigationBarItem(ITextVersion textVersion, RoslynNavigationBarI
             UnderlyingItem = underlyingItem;
         }
 
-        private static ImmutableArray<TextSpan> GetSpans(RoslynNavigationBarItem underlyingItem)
+        private static ImmutableArray<TextSpan> GetInDocumentSpans(RoslynNavigationBarItem underlyingItem)
         {
-            return underlyingItem is RoslynNavigationBarItem.SymbolItem symbolItem && symbolItem.Location.InDocumentInfo != null
-                ? symbolItem.Location.InDocumentInfo.Value.spans
-                : ImmutableArray<TextSpan>.Empty;
-        }
-
-        private static TextSpan? GetNavigationSpan(RoslynNavigationBarItem underlyingItem)
-        {
-            return underlyingItem is RoslynNavigationBarItem.SymbolItem symbolItem && symbolItem.Location.InDocumentInfo != null
-                ? symbolItem.Location.InDocumentInfo.Value.navigationSpan
-                : null;
+            return underlyingItem switch
+            {
+                // For a regular symbol we want to select it if the user puts their caret in any of the spans of it in this file.
+                RoslynNavigationBarItem.SymbolItem { Location.InDocumentInfo: { } symbolInfo } => symbolInfo.spans,
+
+                // An actionless item represents something that exists just to show a child-list, but should otherwise
+                // not navigate or cause anything to be generated.  However, we still want to automatically select it whenever
+                // the user puts their caret in any of the spans of its child items in this file.
+                RoslynNavigationBarItem.ActionlessItem actionless => actionless.ChildItems.SelectMany(i => GetInDocumentSpans(i)).OrderBy(s => s.Start).Distinct().ToImmutableArray(),
```

Comment:
this is the primary location of the fix.  We use GetInDocumentSpans to report all the places in the file that should be considered associated with a particular nav bar item.  In the case of actionlessitem, we were effectively stating we though it had no locations of interest, and so we would not select it when you moved into any of the corresponding items.